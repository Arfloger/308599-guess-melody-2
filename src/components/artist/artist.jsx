import React from 'react';
import PropTypes from 'prop-types';

export const Artist = (props) => {
  const {value, srcPath, title} = props;
  return (
    <div className="artist">
      <input
        className="artist__input visually-hidden"
        type="radio"
        name="answer"
        value={value}
        id={value}
      />
      <label className="artist__name" htmlFor={value}>
        <img
          className="artist__picture"
          src={srcPath}
          alt={title}/>
        {title}
      </label>
    </div>
  );
};

Artist.propTypes = {
  value: PropTypes.string.isRequired,
  srcPath: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
