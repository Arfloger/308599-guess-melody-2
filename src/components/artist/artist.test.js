import React from "react";
import renderer from "react-test-renderer";

import {Artist} from "./artist.jsx";

it(`Artist correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<Artist
      value={``}
      srcPath={``}
      title={``}
    />).toJSON();

  expect(tree).toMatchSnapshot();
});
