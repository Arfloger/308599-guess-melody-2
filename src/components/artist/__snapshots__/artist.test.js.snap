// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist correctly renders after relaunch 1`] = `
<div
  className="artist"
>
  <input
    className="artist__input visually-hidden"
    id=""
    name="answer"
    type="radio"
    value=""
  />
  <label
    className="artist__name"
    htmlFor=""
  >
    <img
      alt=""
      className="artist__picture"
      src=""
    />
    
  </label>
</div>
`;
